go/vt/tabletmanager/agent_rpc_actions.go

```diff
@@ -466,7 +466,7 @@ func (agent *ActionAgent) DemoteMaster(ctx context.Context) (myproto.Replication
 	// database. This will in effect close the connection pools to the
 	// database.
 	tablet := agent.Tablet()
-	agent.disallowQueries(tablet.Tablet, "DemoteMaster marks server rdonly")
+	agent.disallowQueries(tablet.Tablet.Type, "DemoteMaster marks server rdonly")
```

Comment:
We should be checking the error here, right? We shouldn't continue if we failed to stop in-flight transactions.